src/core/src/package_managers/TdnfPackageManager.py

```diff
@@ -640,6 +636,69 @@ def update_os_patch_configuration_sub_setting(self, patch_configuration_sub_sett
+        # type () -> None
+        self.__init_auto_update_for_dnf_automatic()
+        self.composite_logger.log("[TDNF] Reverting the current automatic OS patch state on the machine to it's system default value for [Service={0}]".format(str(self.current_auto_os_update_service)))
+        is_service_installed, enable_on_reboot_value, download_updates_value, apply_updates_value = self.__get_current_auto_os_updates_setting_on_machine()
+
+        if not is_service_installed:
+            self.composite_logger.log_debug("[TDNF] Machine default auto OS update service is not installed on the VM and hence no config to revert. [Service={0}]".format(str(self.current_auto_os_update_service)))
+            return
+
+        self.composite_logger.log_debug("[TDNF] Logging current configuration settings for auto OS updates [Service={0}][Is_Service_Installed={1}][Machine_default_update_enable_on_reboot={2}][{3}={4}]][{5}={6}]"
+                                        .format(str(self.current_auto_os_update_service), str(is_service_installed), str(enable_on_reboot_value), str(self.download_updates_identifier_text), str(download_updates_value), str(self.apply_updates_identifier_text), str(apply_updates_value)))
+
+        image_default_patch_configuration_backup = self.__get_image_default_patch_configuration_backup()
+        self.composite_logger.log_debug("[TDNF] Logging system default configuration settings for auto OS updates. [Settings={0}]".format(str(image_default_patch_configuration_backup)))
+        is_backup_valid = self.is_image_default_patch_configuration_backup_valid(image_default_patch_configuration_backup)
+
+        if is_backup_valid:
+            download_updates_value_from_backup = image_default_patch_configuration_backup[self.current_auto_os_update_service][self.download_updates_identifier_text]
+            apply_updates_value_from_backup = image_default_patch_configuration_backup[self.current_auto_os_update_service][self.apply_updates_identifier_text]
+            enable_on_reboot_value_from_backup = image_default_patch_configuration_backup[self.current_auto_os_update_service][self.enable_on_reboot_identifier_text]
+
+            self.update_os_patch_configuration_sub_setting(self.download_updates_identifier_text, download_updates_value_from_backup, self.auto_update_config_pattern_match_text)
+            self.update_os_patch_configuration_sub_setting(self.apply_updates_identifier_text, apply_updates_value_from_backup, self.auto_update_config_pattern_match_text)
+            if str(enable_on_reboot_value_from_backup).lower() == 'true':
+                self.enable_auto_update_on_reboot()
+        else:
+            self.composite_logger.log_debug("[TDNF] Since the backup is invalid or does not exist for current service, we won't be able to revert auto OS patch settings to their system default value. [Service={0}]".format(str(self.current_auto_os_update_service)))
```

Comment:
Under what conditions could this not qualify for log_error? (without affecting operation state)

Reply:
Especially since reading the backup is also a log_error

Reply:
There's a special case, where a VM does not contain any image default config or Image default OS updates are not enabled.
Since Image Default updates are not enabled, our initial patch state checks computes current_auto_os_patch_state=Disabled. In this case, we do not explicitly disable/change anything on the VM, thereby not logging anything in backup file.

Now, if patch_mode is later set to 'ImageDefault', this code path will not find any backup file since no changes were ever made in OS updates on the VM. Here, the log is just a statement for awareness rather than an error